File: android/app/src/main/java/be/sportgreenmoove/app/ui/OptionsScreen.kt
package be.sportgreenmoove.app.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import be.sportgreenmoove.app.design.Sgm
import be.sportgreenmoove.app.design.SgmColor
import be.sportgreenmoove.app.design.SgmRadius
import be.sportgreenmoove.app.design.SgmType

private val OptionSections = listOf(
    OptionSection(
        "SÉCURITÉ",
        listOf(
            OptionRow(SgmIcon.Profile, "Contact urgence", "+32 470 00 00 00", "Action visible pendant une course active"),
            OptionRow(SgmIcon.Check, "Journal d'audit", "Activé", "Pickup, dropoff, consentements et paiements"),
            OptionRow(SgmIcon.Groups, "Consentement parent", "Obligatoire", "Chaque enfant reste lié à son tuteur"),
        ),
    ),
    OptionSection(
        "LOCALISATION",
        listOf(
            OptionRow(SgmIcon.Location, "Arrière-plan", "Trajets actifs", "Suivi arrêté hors session de course"),
            OptionRow(SgmIcon.Search, "Alertes position", "Activées", "Dernière mise à jour et statut obsolète"),
            OptionRow(SgmIcon.Bell, "Notifications", "Activées", "Départ, pickup, dropoff et retard"),
        ),
    ),
    OptionSection(
        "PAIEMENTS",
        listOf(
            OptionRow(SgmIcon.Award, "Carte", "Stripe PaymentSheet", "Autorisation avant validation conducteur"),
            OptionRow(SgmIcon.ArrowRight, "Payout", "Stripe Connect", "Versements visibles dans le ledger"),
            OptionRow(SgmIcon.Settings, "Frais plateforme", "0 EUR", "En attente des conditions business"),
        ),
    ),
)

@Suppress("UNUSED_PARAMETER")
@Composable
fun OptionsScreen(onBack: () -> Unit) {
    V2Screen {
        V2TopBar("OPTIONS")
        OptionHero()
        OptionSections.forEach { section ->
            V2SectionLabel(section.title)
            OptionSectionCard(section.rows)
        }
        Box(Modifier.height(76.dp))
    }
}

@Composable
private fun OptionHero() {
    Column(
        modifier = Modifier
            .padding(start = 20.dp, top = 4.dp, end = 20.dp, bottom = 16.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(Sgm.colors.bgCard)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(20.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Text("CONTRÔLES DU COMPTE", style = SgmType.Eyebrow.copy(color = SgmColor.Green, fontSize = 11.sp, letterSpacing = 0.14.em))
        Text("Sécurité, localisation et paiements", style = SgmType.DisplayLG.copy(color = Sgm.colors.textPrimary, fontSize = 20.sp, letterSpacing = 0.04.em))
        Text(
            "Les réglages sensibles restent liés aux courses actives et aux consentements tuteur.",
            style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 12.sp, fontWeight = FontWeight.Medium),
        )
    }
}

@Composable
private fun OptionSectionCard(rows: List<OptionRow>) {
    Column(
        modifier = Modifier
            .padding(start = 20.dp, end = 20.dp, bottom = 16.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.LG))
            .background(Sgm.colors.bgSurface)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.LG)),
    ) {
        rows.forEachIndexed { index, row ->
            OptionRowView(row, showDivider = index < rows.lastIndex)
        }
    }
}

@Composable
private fun OptionRowView(row: OptionRow, showDivider: Boolean) {
    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 13.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            SgmLineIcon(row.icon, tint = Sgm.colors.textMuted, modifier = Modifier.size(18.dp))
            Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    Text(row.label, style = SgmType.BodySM.copy(color = Sgm.colors.textPrimary, fontSize = 14.sp, fontWeight = FontWeight.SemiBold), modifier = Modifier.weight(1f))
                    Text(row.value, style = SgmType.BodyXS.copy(color = SgmColor.Green, fontSize = 12.sp, fontWeight = FontWeight.Bold))
                }
                Text(row.detail, style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 11.sp, fontWeight = FontWeight.Medium))
            }
        }
        if (showDivider) Box(Modifier.padding(start = 46.dp).fillMaxWidth().height(1.dp).background(Sgm.colors.border))
    }
}

private data class OptionSection(val title: String, val rows: List<OptionRow>)
private data class OptionRow(val icon: SgmIcon, val label: String, val value: String, val detail: String)

File: ios/Sources/SportsGreenMooveApp/OptionsScreen.swift
import SwiftUI

struct OptionsScreen: View {
    @Environment(AppState.self) private var appState
    @State private var reason = "Sécurité"
    @State private var description = ""
    @State private var emergency = false
    @State private var submitting = false
    @State private var message: String?

    var body: some View {
        SGMScreen {
            SGMTopBar(title: "OPTIONS", showsBack: true)
            OptionsHero()
            SGMSectionLabel("SÉCURITÉ")
            OptionsSectionCard(rows: [
                .init(icon: .profile, label: "Contact urgence", value: "+32 470 00 00 00", detail: "Action visible pendant une course active"),
                .init(icon: .check, label: "Journal d'audit", value: "Activé", detail: "Pickup, dropoff, consentements et paiements"),
                .init(icon: .groups, label: "Consentement parent", value: "Obligatoire", detail: "Chaque enfant reste lié à son tuteur"),
            ])
            SGMSectionLabel("LOCALISATION")
            OptionsSectionCard(rows: [
                .init(icon: .location, label: "Arrière-plan", value: "Trajets actifs", detail: "Suivi arrêté hors session de course"),
                .init(icon: .search, label: "Alertes position", value: "Activées", detail: "Dernière mise à jour et statut obsolète"),
                .init(icon: .bell, label: "Notifications", value: "Activées", detail: "Départ, pickup, dropoff et retard"),
            ])
            SGMSectionLabel("PAIEMENTS")
            OptionsSectionCard(rows: [
                .init(icon: .award, label: "Carte", value: "Stripe PaymentSheet", detail: "Autorisation avant validation conducteur"),
                .init(icon: .arrowRight, label: "Payout", value: "Stripe Connect", detail: "Versements visibles dans le ledger"),
                .init(icon: .settings, label: "Frais plateforme", value: "0 EUR", detail: "En attente des conditions business"),
            ])
            SGMSectionLabel("SUPPORT")
            reportCard
        }
    }

    private var reportCard: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("SIGNALER AU SUPPORT")
                .font(.sgmDisplay(18))
                .tracking(.sgmWide(for: 18))
                .foregroundStyle(SGM.textPrimary)
            ReportField(label: "Raison", text: $reason, axis: .horizontal)
            ReportField(label: "Description", text: $description, axis: .vertical)
            Toggle("Urgent", isOn: $emergency)
                .font(.sgmBody(13, weight: .bold))
                .foregroundStyle(SGM.textPrimary)
                .tint(SGM.green)
            SGMButton(title: submitting ? "ENVOI..." : "ENVOYER", action: { Task { await submitReport() } })
            if let message {
                Text(message)
                    .font(.sgmBody(12, weight: .bold))
                    .foregroundStyle(message.hasPrefix("Signalement") ? SGM.green : SGM.orange)
            }
        }
        .padding(14)
        .background(SGM.bgSurface, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
        .padding(.horizontal, SGMSpace.padScreen)
    }

    private func submitReport() async {
        guard !submitting else { return }
        let cleanReason = reason.trimmingCharacters(in: .whitespacesAndNewlines)
        let cleanDescription = description.trimmingCharacters(in: .whitespacesAndNewlines)
        guard cleanReason.count >= 2, cleanDescription.count >= 5 else {
            message = "Ajoutez une raison et une description précise."
            return
        }

        submitting = true
        message = nil
        defer { submitting = false }
        do {
            let reportId = try await appState.firebase.createReport(
                subjectType: "other",
                subjectId: nil,
                reason: cleanReason,
                description: cleanDescription,
                emergency: emergency
            )
            description = ""
            message = "Signalement envoyé: \(reportId)"
        } catch {
            message = error.localizedDescription
        }
    }
}

private struct OptionsHero: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text("CONTRÔLES DU COMPTE")
                .font(.sgmBody(11, weight: .bold))
                .foregroundStyle(SGM.green)
            Text("Sécurité, localisation et paiements")
                .font(.sgmDisplay(20))
                .tracking(.sgmWide(for: 20))
                .foregroundStyle(SGM.textPrimary)
            Text("Les réglages sensibles restent liés aux courses actives et aux consentements tuteur.")
                .font(.sgmBody(12, weight: .medium))
                .foregroundStyle(SGM.textMuted)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: 20, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 20, style: .continuous).stroke(SGM.border, lineWidth: 1))
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.bottom, 8)
    }
}

private struct ReportField: View {
    let label: String
    @Binding var text: String
    var axis: Axis

    var body: some View {
        TextField(label, text: $text, axis: axis)
            .font(.sgmBody(14, weight: .medium))
            .foregroundStyle(SGM.textPrimary)
            .lineLimit(axis == .vertical ? 3...5 : 1...1)
            .padding(.horizontal, 12)
            .padding(.vertical, 10)
            .background(SGM.bgInput, in: RoundedRectangle(cornerRadius: SGMRadius.md, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: SGMRadius.md, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

private struct OptionsSectionCard: View {
    let rows: [OptionsRow]

    var body: some View {
        VStack(spacing: 0) {
            ForEach(rows) { row in
                OptionsRowView(row: row)
                if row.id != rows.last?.id {
                    Divider().background(SGM.border).padding(.leading, 46)
                }
            }
        }
        .background(SGM.bgSurface, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.bottom, 8)
    }
}

private struct OptionsRowView: View {
    let row: OptionsRow

    var body: some View {
        HStack(spacing: 12) {
            SGMIconView(icon: row.icon, size: 18, color: SGM.textMuted)
            VStack(alignment: .leading, spacing: 2) {
                HStack(spacing: 8) {
                    Text(row.label)
                        .font(.sgmBody(14, weight: .semibold))
                        .foregroundStyle(SGM.textPrimary)
                    Spacer()
                    Text(row.value)
                        .font(.sgmBody(12, weight: .bold))
                        .foregroundStyle(SGM.green)
                        .lineLimit(1)
                        .minimumScaleFactor(0.72)
                }
                Text(row.detail)
                    .font(.sgmBody(11, weight: .medium))
                    .foregroundStyle(SGM.textMuted)
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 13)
    }
}

private struct OptionsRow: Identifiable {
    let id = UUID()
    let icon: SGMIcon
    let label: String
    let value: String
    let detail: String
}
